scarb/src/compiler/plugin/proc_macro/compilation.rs

```diff
@@ -60,10 +64,81 @@ pub fn check_unit(unit: ProcMacroCompilationUnit, ws: &Workspace<'_>) -> Result<
     run_cargo(CargoAction::Check, &package, ws)
 }
 
-pub fn fetch_package(package: &Package, ws: &Workspace<'_>) -> Result<()> {
+fn get_cargo_package_name(package: &Package) -> String {
+    let cargo_toml_path = package.root().join(CARGO_MANIFEST_FILE_NAME);
+
+    let cargo_toml: toml::Value =
+        toml::from_str(&fs::read_to_string(cargo_toml_path).expect("Could not read `Cargo.toml`."))
+            .expect("Could not convert `Cargo.toml` to toml.");
+
+    let package_section = cargo_toml
+        .get("package")
+        .expect("Could not get package section from `Cargo.toml`.");
+
+    let package_name = package_section
+        .get("name")
+        .expect("Could not get name field from `Cargo.toml`.")
+        .as_str()
+        .unwrap();
+
+    package_name.to_string()
+}
+
+fn get_cargo_library_name(package: &Package) -> String {
```

Comment:
We can also show a warning here.

Reply:
Added in [f58b098](https://github.com/software-mansion/scarb/pull/1605/commits/f58b098eb18a6145a762df9cb94637b0a5a57875)